platform/conf/conf.go

```diff
@@ -143,30 +147,34 @@ func (u *UserData) Render() (*Conf, error) {
 		// pass through scripts unmodified, you are on your own.
 		c.script = u.data
 	case kindIgnition:
-		// Reports collapse errors to their underlying strings
-		haveEntry := func(report report.Report, err error) bool {
-			for _, entry := range report.Entries {
-				if err.Error() == entry.Message {
-					return true
-				}
-			}
-			return false
+		ver, err := ign.Version([]byte(u.data))
+		// process indeterminable configs with the current version
+		// so we can get parse errors
+		if err != nil && err != ign.ErrVersionIndeterminable {
+			return nil, err
 		}
 
-		ignc, report, err := v2.Parse([]byte(u.data))
-		if err == nil {
+		switch ver {
+		default:
+			ignc, report, err := v21.Parse([]byte(u.data))
+			if err != nil {
+				plog.Errorf("invalid userdata: %v", report)
+				return nil, err
+			}
+			c.ignitionV21 = &ignc
```

Comment:
Possible for which to happen?  We end up in this switch arm for all Ignition 2.1 configs, and for any config that `Version` can't parse.